### DISCUSSION TITLE
Performance has regressed significantly in version 0.10.0 compared to version 0.8.1

### DISCUSSION THREAD

**Discussion by xuchaoqian** [tqwewe/kameo#41] [General]
See details [here](https://github.com/xuchaoqian/rust-actors-benches/commit/92d475f32854a8f2332541edd58cb8edcc5f1859#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R33), and the summary is as follows:

![image](https://github.com/user-attachments/assets/23306554-c6d5-4bfe-b82b-f1bf25bfcc43)

**Comment by tqwewe** [comment]
Thanks for bringing this up, its quite interesting, I can't think of anything obvious that might've caused this. I'll do some investigating into what might've caused it and get back to you

**Reply by tqwewe** [reply]
Okay I've done some binary searching through the commits since 0.8 and benchmarks, and found a performance degredation in `tell` multi-threaded, and `spawn`.

So far I've only investigated the `tell` degredation and found it to be an unecessary clone of mailboxes. I'll push some updates which makes the mailbox borrowed in requests.

![Creation of 10,000 actors vs  Commit](https://github.com/user-attachments/assets/db78264a-0270-40ee-9830-43f5cb97fdcb)
![Tell 10,000 messages (multi-threaded) vs  Commit](https://github.com/user-attachments/assets/8dfcdb47-64a0-4387-a207-a37cb07bc625)
![Ask 10,000 messages (single-threaded) vs  Commit](https://github.com/user-attachments/assets/c79a5765-ac6d-4d7b-9941-29d03b5a7599)

**Reply by tqwewe** [reply]
I've pushed another commit which should solve the performance issues with actor spawning.
https://github.com/tqwewe/kameo/pull/43

**Reply by tqwewe** [reply]
@xuchaoqian v0.11.0 is out which should improve some of these performance issues, thanks again for creating this discussion!

**Reply by xuchaoqian** [reply]
@tqwewe Thank you for your great work! I have updated the bench results with the latest versions of all libraries.

Kameo is extremely fast at creating actors and sending messages.
In particular, it is much faster than Actix for asking messages in single-threaded mode, and also faster than Actix for telling messages in multi-threaded/single-threaded modes.

The only downside of Kameo is that it is slower than Actix at creating actors, approximately 2 to 3 times slower.
